tests/Unit/Domain/FunctionsOfTime/Test_IntegratedFunctionOfTime.cpp

```diff
@@ -0,0 +1,170 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+#include <limits>
+#include <memory>
+#include <random>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/IntegratedFunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/RegisterDerivedWithCharm.hpp"
+#include "Domain/FunctionsOfTime/ThreadsafeList.tpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Serialization/Serialize.hpp"
+
+namespace domain {
+namespace {
+void test(const gsl::not_null<FunctionsOfTime::FunctionOfTime*> f_of_t,
```

Comment:
Define this in the function instead of taking it as an argument, since the function won't work if it isn't pointing to the same object as the `f_of_t_derived` argument.  (Or take this one and do the cast to get the other.)